cpp/src/copying/concatenate.cu

```diff
@@ -374,11 +374,18 @@ void traverse_children::operator()<cudf::string_view>(host_span<column_view cons
   size_t const total_char_count = std::accumulate(
     cols.begin(), cols.end(), std::size_t{}, [stream](size_t a, auto const& b) -> size_t {
       strings_column_view scv(b);
-      return a + (b.is_empty()
-                    ? 0
-                    : cudf::detail::get_value<offset_type>(
-                        scv.offsets(), scv.offset() + b.size(), stream) -
-                        cudf::detail::get_value<offset_type>(scv.offsets(), scv.offset(), stream));
+      return a + (scv.is_empty() ? 0
```

Comment:
Not urgent but this feels ripe for refactoring to a lambda given we have dual nested ternary statements. 

something like 

```cxx
auto computed_length = [](...)
{
....
}; 
return a + (scv.is_empty() ? 0 : computed_length);